### ISSUE TITLE
Unexpected method 'appcast' called on Cask haskell-platform

### ISSUE THREAD

**Issue by biztrology-kd** [Homebrew/homebrew-cask#180672]
### Verification

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [X] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match / SHA256 mismatch](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match) error (do not open an issue before trying to open a PR to fix first).

### Description of issue

 Unexpected method 'appcast' called on Cask haskell-platform.

### Command that failed

brew upgrade

### Output of command with `--verbose --debug`

```shell
brew upgrade --verbose --debug
/usr/local/Homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromInstalledPathLoader): loading haskell-platform
Error: Unexpected method 'appcast' called on Cask haskell-platform.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
/usr/local/Homebrew/Library/Homebrew/brew.rb (Cask::CaskLoader::FromAPILoader): loading react-native-debugger
```

### Output of `brew doctor` and `brew config`

```shell
brew config:
% brew config
HOMEBREW_VERSION: 4.3.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: c3ae0327c01aab461531bfc0fcdfb4dd3a779205
Last commit: 7 days ago
Core tap JSON: 26 Jul 00:56 UTC
Core cask tap JSON: 26 Jul 00:56 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BOTTLE_DOMAIN: https://mirrors.tuna.tsinghua.edu.cn/homebrew-bottles
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 14.0.0 build 1400
Git: 2.45.2 => /usr/local/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 12.7.5-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2

brew doctor:

Error: Unexpected method 'appcast' called on Cask haskell-platform.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
Error: Unexpected method 'appcast' called on Cask haskell-platform.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

```

### Output of `brew tap`

```shell
dart-lang/dart
```

**Comment by khipp** [comment]
Take a look at our pinned issue: https://github.com/Homebrew/homebrew-cask/issues/173201#issuecomment-2101662209.

**Comment by SMillerDev** [comment]
Duplicate of https://github.com/Homebrew/homebrew-cask/issues/173201
